ui/src/app/utils.ts

```diff
@@ -38,62 +39,107 @@ export const oidToName = (oid: string) => {
-    }
+function hexToIp(hex: ArrayBuffer): string {
+    const byteArray = new Uint8Array(hex);
+    return Array.from(byteArray).map(byte => byte.toString(10)).join('.');
+}
+
+function parseExtensions(extensions: Extensions) {
+    const sansDns: string[] = [];
+    const sansIp: string[] = [];
+    let is_ca = false;
+
+    extensions.extensions.forEach(extension => {
+        let extensionName: string;
+        try {
+            extensionName = oidToName(extension.extnID);
+        } catch (error) {
+            console.error(`Unrecognized extension OID: ${extension.extnID}`);
+            return;
+        }
+
+        if (extensionName === "Subject Alternative Name") {
+            extension.parsedValue.altNames.forEach((altName: { type: number; value: any; }) => {
+                if (altName.type == 2) {
+                    sansDns.push(altName.value);
+                } else if (altName.type == 7) {
+                    sansIp.push(hexToIp(altName.value.valueBlock.valueHex));
+                }
+            });
+        } else if (extensionName === "Basic Constraint") {
+            console.log(extension);
+            is_ca = extension.parsedValue.cA;
+        }
+    });
 
-    // Load CSR object
+    return { sansDns, sansIp, is_ca };
+}
+
+export const extractCSR = (csrPemString: string) => {
+    const arrayBuffer = pemToArrayBuffer(csrPemString);
+    const asn1 = fromBER(arrayBuffer);
     const csr = new CertificationRequest({ schema: asn1.result });
 
     // Extract subject information from CSR
     const subjects = csr.subject.typesAndValues.map(typeAndValue => ({
         type: oidToName(typeAndValue.type),
         value: typeAndValue.value.valueBlock.value
     }));
+    const getValue = (type: string) => subjects.find(subject => subject.type === type)?.value;
 
-    // Look for extensions attribute in CSR
-    const attributes = csr.attributes?.map(typeAndValue => ({
-        type: oidToName(typeAndValue.type),
-        value: typeAndValue.values
-    }))
-    return { subjects }
+    const commonName = getValue("Common Name");
+    const organization = getValue("Organization Name");
+    const emailAddress = getValue("Email Address");
+    const country = getValue("Country");
+    const locality = getValue("Locality");
+
+    let sansDns: string[] = [];
+    let sansIp: string[] = [];
+    let is_ca = false;
+
+    if (csr.attributes) {
+        const extensionAttributes = csr.attributes.filter(attribute => oidToName(attribute.type) === "Extension Request");
+        if (extensionAttributes.length > 0) {
+            const extensions = new Extensions({ schema: extensionAttributes[0].values[0] });
+            ({ sansDns, sansIp, is_ca } = parseExtensions(extensions));
+        }
+    }
+    return {
+        commonName,
+        organization,
+        emailAddress,
+        country,
+        locality,
+        sansDns,
+        sansIp,
+        is_ca
+    }
 }
 
 export const extractCert = (certPemString: string) => {
```

Comment:
Please create a new ticket and change the description of 276 if you aren't going to tackle extracting certificate information here.

Reply:
Ticket created